test_util/launch.py

```diff
@@ -1,20 +1,282 @@
+
+    def get_stack(self, info):
+        """Returns the correct class interface depending how the AWS CF is configured
+        NOTE: only supports Simple Cloudformation currently
+        """
+        try:
+            return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+        except Exception as ex:
+            raise LauncherError('StackNotFound', '{} is not accessible'.format(info['stack_name'])) from ex
+
+
+def get_launcher(launcher_type, provider_info):
+    """Returns a launcher given a launcher type (string) and a dictionary of
+    appropriate provier_info
+    """
+    if launcher_type == 'cloudformation':
+        check_keys(provider_info, ['region', 'access_key_id', 'secret_access_key'])
+        boto_wrapper = BotoWrapper(
+            provider_info['region'], provider_info['access_key_id'], provider_info['secret_access_key'])
+        return AwsCloudformationLauncher(boto_wrapper)
+    else:
+        raise LauncherError('UnsupportedAction', 'Launcher type not supported: {}'.format(launcher_type))
+
+
+def convert_host_list(host_list):
+    """ Makes Host tuples more readable when using describe
+    """
+    return [{'private_ip': h.private_ip, 'public_ip': h.public_ip} for h in host_list]
+
+
+def check_keys(user_dict, key_list):
+    missing = [k for k in key_list if k not in user_dict]
+    if len(missing) > 0:
+        raise LauncherError('MissingInput', 'The following keys were required but'
+                            'not provided: {}'.format(repr(missing)))
+
+
+def main(argv):
+    args = docopt(__doc__, argv=argv, version=DCOS_LAUNCH_VERSION)
+
+    try:
+        if args['create']:
+            info_path = args['--info-path']
+            if os.path.exists(info_path):
+                raise LauncherError('InputConflict', 'Target info path already exists!')
+            config = load_yaml(args['--config-path'])
+            check_keys(config, ['type', 'provider_info', 'this_is_a_temporary_config_format_do_not_put_in_production'])
+            write_json(info_path, get_launcher(config['type'], config['provider_info']).create(config))
+            return 0
+
+        info = load_json(args['--info-path'])
+        check_keys(info, ['type', 'provider'])
+        launcher = get_launcher(info['type'], info['provider'])
+
+        if args['wait']:
+            launcher.wait(info)
+            print('Cluster is ready!')
+            return 0
+
+        if args['describe']:
+            print(json_prettyprint(launcher.describe(info)))
+            return 0
+
+        if args['test']:
+            if len(args['<test_command>']) > 0:
+                test_cmd = ' '.join(args['<test_command>'])
+            else:
+                test_cmd = 'py.test'
+            launcher.test(info, test_cmd)
+            return 0
 
-def main():
-    docopt(__doc__)
+        if args['delete']:
+            launcher.delete(info)
+            return 0
 
-    raise NotImplementedError()
+    except LauncherError as ex:
```

Comment:
Should also except YamlParseError (https://github.com/dcos/dcos/blob/master/pkgpanda/util.py#L131) which is raied when load_yaml fails to parse a file. 

May also make sense to teach load_yaml to catch https://docs.python.org/3/library/exceptions.html#OSError and re-throw it as a YamlParseError with details like the filename.